File: public/docs/_examples/dependency-injection/dart/web/heroes/hero-list.component.2.dart
// #docregion
import "package:angular2/core.dart";
import "hero.dart";
import "hero.service.dart";

@Component(
    selector: "hero-list",
    template: '''
  <div *ngFor="#hero of heroes">
    {{hero.id}} - {{hero.name}}
  </div>
  ''')
class HeroListComponent {
  List<Hero> heroes;
  //#docregion ctor
  HeroListComponent(HeroService heroService) {
    heroes = heroService.getHeroes();
  }
}

File: public/docs/_examples/rxjs/ts/src/app/hero-list.component.2.ts
// #docplaster
// #docregion
import { Component, OnInit }   from '@angular/core';
import { Observable }          from 'rxjs/Observable';

import { HeroService }   from './hero.service';
import { Hero }          from './hero';

@Component({
// #docregion async-pipe
  template: `
    <h2>HEROES</h2>
    <ul class="items">
      <li *ngFor="let hero of heroes$ | async">
        <span class="badge">{{ hero.id }}</span> {{ hero.name }}
      </li>
    </ul>
  `
// #enddocregion async-pipe
})
// #docregion observable-heroes
export class HeroListComponent implements OnInit {
  heroes$: Observable<Hero[]>;

  constructor(
    private service: HeroService
  ) {}

  ngOnInit() {
    this.heroes$ = this.service.getHeroes();
  }
}
// #enddocregion
